Source/Core/Common/TypeUtils.h

```diff
@@ -82,4 +83,18 @@ static_assert(!IsNOf<int, 1, int, int>::value);
 static_assert(IsNOf<int, 2, int, int>::value);
 static_assert(IsNOf<int, 2, int, short>::value);  // Type conversions ARE allowed
 static_assert(!IsNOf<int, 2, int, char*>::value);
+
+template <typename T>
+concept Enum = std::is_enum_v<T>;
+
+template <typename T, typename Underlying>
+concept TypedEnum = requires
+{
+  {
+    std::underlying_type_t<T>()
+    } -> std::same_as<Underlying>;
```

Comment:
The formatting is dumb because we're on clang-format-13.
clang-format-19 makes this a much more sensible:
```cpp
{ std::underlying_type_t<T>() } -> std::same_as<Underlying>;
```